flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt

```diff
@@ -142,6 +180,60 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
     }
   }
 
+  private fun beginNativeFrames(result: Result) {
+    if (!enableAutoPerformanceTracking) {
+      result.success(null)
+      return
+    }
+    if (framesTracker == null) {
+      framesTracker = ActivityFramesTracker(LoadClass())
+    }
+    activity?.get()?.let {
+      framesTracker?.addActivity(it)
+    }
+    result.success(null)
+  }
+
+  private fun endNativeFrames(id: String?, result: Result) {
+    if (!enableAutoPerformanceTracking) {
+      result.success(null)
+      return
+    }
+    if (id == null) {
+      Log.w("Sentry", "Parameter id cannot be null.")
+      result.success(null)
+      return
+    }
+
+    val activity = activity?.get()
+
+    if (activity == null) {
+      result.success(null)
+      return
+    }
+
+    val sentryId = SentryId(id)
+    framesTracker?.setMetrics(activity, sentryId)
+
+    val metrics = framesTracker?.takeMetrics(sentryId)
+    val total = metrics?.get("frames_total")?.getValue()?.toInt()
+    val slow = metrics?.get("frames_slow")?.getValue()?.toInt()
+    val frozen = metrics?.get("frames_frozen")?.getValue()?.toInt()
+
+    if (total == null || slow == null || frozen == null) {
+      result.success(null)
```

Comment:
By doing: https://github.com/getsentry/sentry-dart/pull/772/files#r825450103
We can avoid this check.